sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -130,7 +130,7 @@ def fillna(self, value, method, axis, **kwargs):
             lambda df,
             value: df.fillna(value, method=method, axis=axis, **kwargs),
             [self._expr, value_expr],
-            preserves_partition_by=partitionings.Singleton(),
+            preserves_partition_by=partitionings.Index(),
```

Comment:
In general, what we previously called `preserves=Singleton()` (or equivalently `preserves=Index()`), i.e. "this expression will preserve any partitioning", now must be indicated by `preserves=Index()`.

Also what we previously called `preserves=Nothing()`, i.e. "this expression cannot preserve any partitioning", is now represented with `preserves=Singleton()`. This is perhaps a little surprising but I think it's easy to reason about, just think of it as "the only thing this expression preserves is Singleton() parititioning, which is trivially preserved".

All the `preserves=` updates are applying this mapping